File: ccif/src/java/other/org/himalay/ccif/HostPropertiesReq.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:56 EST 2016")

public  class HostPropertiesReq extends   APDUBaseFactory.APDUBase { //Concrete type is HostPropertiesReq




    // members variables
    // header
    public APDUHeader header ;
    // length
    public short length ;
    // hpr
    public ArrayList<HostPropertiesReqMsg> hpr ;

    public HostPropertiesReq () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize length

        // Initialize hpr
        hpr= new ArrayList<HostPropertiesReqMsg>();
        hpr.setMemberSize(0);
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        preRead();
        int retVal= 0;
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read hpr
        for (; istream.available() > 0 ; ) {
            HostPropertiesReqMsg temp;
            temp = new HostPropertiesReqMsg();
            retVal += temp.read(istream);
            hpr.add(temp);
        }

        postRead();
        return retVal;
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read header
        retVal += header.read(istream);
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read hpr
        for (; istream.available() > 0 ; ) {
            HostPropertiesReqMsg temp;
            temp = new HostPropertiesReqMsg();
            retVal += temp.read(istream);
            hpr.add(temp);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for header **/
        }

        {   /** fix dependent sizes for hpr**/
        }

        // write header
        if (header!=null)retVal +=header.write(ostream);
        // write length
        ostream.writeByte(length);
        retVal +=1;
        // write hpr
        {
            ArrayList<HostPropertiesReqMsg> temp1 = hpr;
            for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++) {
                HostPropertiesReqMsg temp2    =    temp1.get(iIdx);
                if (temp2!=null)retVal +=temp2.write(ostream);
            }
        }
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("HostPropertiesReq\n");
        dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {
            dc.indent();
            dc.getPs().println("header") ;
            retVal +=header.dump(dc);
        }
        // write length
        dc.indent();
        dc.getPs().println("length="+length+"(0x"+ Integer.toHexString(length) + ")" );
        // write hpr
        {
            ArrayList<HostPropertiesReqMsg> temp1 = hpr;
            for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++) {
                HostPropertiesReqMsg element    = temp1.get(iIdx);
                dc.indent();
                dc.getPs().println(iIdx);
                if ( element != null ) {
                    dc.indent();
                    dc.getPs().println("element") ;
                    retVal +=element.dump(dc);
                }
            }
        }
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}


    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for length
    //public short getLength()
    //{
    //    return length ;
    //}


    // Setter for length
    //public void setLength(short val)
    //{
    //    this.length= val;
    //}
    // Getter for hpr
    //public ArrayList<HostPropertiesReqMsg> getHpr()
    //{
    //    return hpr ;
    //}


    // Setter for hpr
    //public void setHpr(ArrayList<HostPropertiesReqMsg> val)
    //{
    //    this.hpr= val;
    //}


    public int addToHpr(HostPropertiesReqMsg val)
    {
        hpr.add(val);
        return hpr.size();
    }


    public int removeFromHpr(HostPropertiesReqMsg val)
    {
        hpr.remove(val);
        return hpr.size();
    }


    public int removeNthFromHpr(int idx)
    {
        hpr.remove(idx);
        return hpr.size();
    }


    public int emptyHpr(int idx)
    {
        hpr.clear();
        return hpr.size();
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }


    public void setHeader(APDUHeader header)
    {
        this.header= header;
    }

    public APDUHeader getHeader()
    {
        return this.header;
    }

}

// End of code

File: libccif/src/bp/HostPropertiesReq.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:14 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "HostPropertiesReq.h"

#ifdef org_himalay_ccif_HostPropertiesReq__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
HostPropertiesReq::HostPropertiesReq() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // length
    length= (ui8)0;
    // hpr
    hpr.setMemberSize(0);
}
// Destructor
HostPropertiesReq::~HostPropertiesReq() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // length

    // hpr
    {
#ifndef org_himalay_ccif_HostPropertiesReqMsg_USE_SMART_PTR
        hpr.deleteMembers();
#endif
    }
}

int HostPropertiesReq::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // hpr
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_HostPropertiesReqMsg__PTR_TYPE temp;
        org_himalay_ccif_HostPropertiesReqMsg__NEW( temp);
        retVal += temp->read(istream);
        hpr.add(temp);
    }
    return retVal;
}

int HostPropertiesReq::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // read hpr
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_HostPropertiesReqMsg__PTR_TYPE temp;
        org_himalay_ccif_HostPropertiesReqMsg__NEW( temp);
        retVal += temp->read(istream);
        hpr.add(temp);
    }

    return retVal;
}

int HostPropertiesReq::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }

    {   /** fix dependent sizes for hpr**/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write length
    ostream.writeUI8(length);
    retVal +=1;
    // write hpr
{   for (ArrayList_iterator(HostPropertiesReqMsg) it=  hpr.getIterator() ; it.hasNext();) {
            {
                retVal +=it.get()->write(ostream);
            }
        }
    }

    return retVal;
}

int HostPropertiesReq::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("HostPropertiesReq\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write length
    dc.indent();
    dc.getPs().print("length=");
    dc.getPs().println((long)length);
    // write hpr
{   for (ArrayList_iterator(HostPropertiesReqMsg) it=  hpr.getIterator() ; it.hasNext();) {
            {
                dc.indent();
                dc.getPs().println("element");
                retVal +=it.get()->dump(dc);
            }
        }
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE HostPropertiesReq::getHeader()
//{
//return this->header;
//}

// Setter for header
//void HostPropertiesReq::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for length
//ui8 HostPropertiesReq::getLength()
//{
//return this->length;
//}

// Setter for length
//void HostPropertiesReq::setLength(ui8 val)
//{
//this->length= val;
//}

// Getter for hpr
//ArrayList(HostPropertiesReqMsg) HostPropertiesReq::getHpr()
//{
//return this->hpr;
//}

// Setter for hpr
//void HostPropertiesReq::setHpr(ArrayList(HostPropertiesReqMsg) val)
//{
//this->hpr= val;
//}

int HostPropertiesReq::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void HostPropertiesReq::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE header)
{
    this->header= header;
}

org_himalay_ccif_APDUHeader__PTR_TYPE HostPropertiesReq::getHeader()
{
    return this->header;
}

#ifdef org_himalay_ccif_HostPropertiesReq__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::HostPropertiesReq* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::HostPropertiesReq* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code